strongbox-web-core/src/main/java/org/carlspring/strongbox/config/SecurityConfig.java

```diff
@@ -25,74 +32,131 @@
     }
 
-    @PostConstruct
-    public void init()
+    /**
+     * This configuration specifies BasicAuthentication rules. Such authentication triggered first and only for concrete
+     * set of URL patterns. All the other authentication configured in {@link JwtSecurityConfig}
+     * 
+     * @author Sergey Bespalov
+     *
+     */
+    @Configuration
+    @Order(1)
+    public static class BasicSecurityConfig extends WebSecurityConfigurerAdapter
     {
-        authorizationConfigProvider.getConfig().ifPresent((config) -> {
-            anonymousAuthenticationFilter.getAuthorities()
-                                         .addAll(config.getAnonymousAuthorities());
-        });
-    }
 
-    @Bean
-    public AnonymousAuthenticationFilter anonymousAuthenticationFilter()
-    {
-        return anonymousAuthenticationFilter;
+        @Override
+        protected void configure(HttpSecurity http)
+            throws Exception
+        {
+            http.sessionManagement()
+                .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                .and()
+                .requestMatchers()
+                .antMatchers("/users/user/authenticate", "/storages")
+                .and()
+                .authorizeRequests()
+                .anyRequest()
+                .authenticated()
+                .and()
+                .httpBasic()
+                .and()
+                .csrf()
+                .disable();
+        }
     }
 
-    @Override
-    protected void configure(HttpSecurity http)
-        throws Exception
+    /**
+     * This configuration specifies JWTAuthentication and Anonymous access rules. It triggered after
+     * {@link BasicSecurityConfig} and authenticate unauthorized requests if needed.
+     * 
+     * @author Sergey Bespalov
+     *
+     */
+    @Configuration
+    @Order(2)
+    public static class JwtSecurityConfig extends WebSecurityConfigurerAdapter
     {
-        http.sessionManagement()
-            .sessionCreationPolicy(SessionCreationPolicy.ALWAYS)
-            .and()
-            .authorizeRequests()
-            .antMatchers("/docs/**", "/assets/**")
-            .permitAll()
-            .and()
-            .anonymous()
-            .authenticationFilter(anonymousAuthenticationFilter)
-            .and()
-            .authorizeRequests()
-            .anyRequest()
-            .authenticated()
-            .and()
-            .httpBasic()
-            .and()
-            .csrf()
-            .disable()
-            .formLogin()
-            .and()
-            .logout()
-            .logoutUrl("/logout");
+
+        @Autowired
+        private AuthorizationConfigProvider authorizationConfigProvider;
+        private AnonymousAuthenticationFilter anonymousAuthenticationFilter;
+
+        public JwtSecurityConfig()
+        {
+            anonymousAuthenticationFilter = new CustomAnonymousAuthenticationFilter("strongbox-unique-key",
```

Comment:
I can't say anything about that as it was done before I have joined the strongbox team.